src/webgl/3d_primitives.js

```diff
@@ -1615,7 +1665,48 @@ p5.RendererGL.prototype.bezierVertex = function(...args) {
       w_x = [this.immediateMode._bezierVertex[0], args[0], args[3], args[6]];
       w_y = [this.immediateMode._bezierVertex[1], args[1], args[4], args[7]];
       w_z = [this.immediateMode._bezierVertex[2], args[2], args[5], args[8]];
+      // Calculate intermediate colors
+      let d0 = Math.sqrt(
+        Math.pow(w_x[0]-w_x[1],2) +
+        Math.pow(w_y[0]-w_y[1],2) +
+        Math.pow(w_z[0]-w_z[1],2)
+      );
+      let d1 = Math.sqrt(
+        Math.pow(w_x[1]-w_x[2],2) +
+        Math.pow(w_y[1]-w_y[2],2) +
+        Math.pow(w_z[1]-w_z[2],2)
+      );
+      let d2 = Math.sqrt(
+        Math.pow(w_x[2]-w_x[3],2) +
+        Math.pow(w_y[2]-w_y[3],2) +
+        Math.pow(w_z[2]-w_z[3],2)
+      );
+      const totalLength = d0 + d1 + d2;
+      d0 /= totalLength;
+      d2 /= totalLength;
+      const firstFillColor = [];
+      const secondFillColor = [];
+      for (k = 0; k < 4; k++) {
+        firstFillColor.push(
+          lastFillColor[k] * (1-d0) + finalFillColor[k] * d0
+        );
+        secondFillColor.push(
+          lastFillColor[k] * d2 + finalFillColor[k] * (1-d2)
+        );
+      }
       for (i = 0; i < LUTLength; i++) {
+        // Interpolate colors using control points
+        this.curFillColor = [0, 0, 0, 0];
+        for (k = 0; k < 4; k++) {
+          this.curFillColor[k] +=
+            this._lookUpTableBezier[i][0] * lastFillColor[k];
```

Comment:
I wonder if there are different names we can use here, as it's maybe a tad unintuitive for readers if "first" is the first *control point* but not the first point in the bezier's hull, and also that it starts with "last" (your explanation for using last instead of previous makes sense, but still might not be what people first assume this variable represents.) Maybe we could even avoid naming each color individually if we just make an array of all four colors? Then `lastFillColor[k]` would be `fillColors[0][k]` (and we could loop over them if we wanted, too.)

Reply:
Certainly, it may not be necessary to prepare four arrays... I thought that the meaning of first might be difficult to understand, so I'm thinking of rewriting it in the form of an array.